### PR TITLE
snippets display improvement

### LINKED ISSUE #648: display improvements to `%sqlcmd snippets`
we need to do some improvements when showing feedback to the user

for context: [snippets.ipynb.zip](https://github.com/ploomber/jupysql/files/11841269/snippets.ipynb.zip)

### PR DESCRIPTION
**Pull request by bbeat2782**
## Describe your changes

Change `%sqlcmd snippets` output into displaying 1. snippet name 2. snippet query 3. snippet table. 

Still working on 1. figuring out how to retrieve `SqlMagic.displaylimit` 2. adding test functions 3. update documentation if needed, but I want to know whether it's on the right direction or not.

## Issue number

Closes #648 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--716.org.readthedocs.build/en/716/

### CODE REVIEW

**Comment by bbeat2782** [comment]
@edublancas 

This is the current output of `%sqlcmd snippets`. Am I on the right track?

Screen Shot 2023-07-06 at 1 14 45 PM

Another question is can we assume `%sqlcmd snippets` will always be called inside a jupyter notebook or do we also think about a user running it in a python script? Asking this question because of figuring out a correct way to get `displaylimit` if the user specifies one.

**Comment by edublancas** [comment]
@bbeat2782 you don't need to query the tables. instead, you can create a table that displays the snippet names

e.g.

```
----------------
| some snippet |
----------------
| another           |
----------------
```

you can use this: https://github.com/ploomber/jupysql/blob/4b938edcdea296dcf1c9585b6ed52fadf50835eb/src/sql/display.py#L10

**Comment by bbeat2782** [comment]
> @bbeat2782 you don't need to query the tables. instead, you can create a table that displays the snippet names
> 
> e.g.
> 
> ```
> ----------------
> | some snippet |
> ----------------
> | another           |
> ----------------
> ```
> 
> you can use this:
> 
> https://github.com/ploomber/jupysql/blob/4b938edcdea296dcf1c9585b6ed52fadf50835eb/src/sql/display.py#L10

@edublancas Oh that is what you meant when you said table. Thanks. Will make the change.

**Comment by bbeat2782** [comment]
Output of `%sqlcmd snippets` when no snippet is stored.
Screen Shot 2023-07-07 at 10 10 08 AM

Output of `%sqlcmd snippets` when chinstrap and chinstrap_sub are stored.
Screen Shot 2023-07-07 at 10 10 11 AM

The above images are screen captured images from [snippet_display_improvement.ipynb.zip](https://github.com/ploomber/jupysql/files/11984363/snippet_display_improvement.ipynb.zip)

File: src/tests/test_magic_cmd.py [hunk 1]
```diff
@@ -416,9 +416,25 @@ def test_test_error(ip, cell, error_type, error_message):
...
+        (None, ["high_price", "high_price_a", "high_price_b"], "invalid"),
+        (
+            "%sqlcmd snippets -d high_price_a",
+            ["high_price", "high_price_b"],
+            "high_price_a",
+        ),
+        ("%sqlcmd snippets -A high_price", ["No snippets stored"], "high_price"),
+    ],
+)
+def test_snippet(ip_snippets, precmd, valid_snippets, invalid_snippet):
```
Comment (edublancas) [inline src/tests/test_magic_cmd.py:431]: we're not testing the returned object. when there are no snippets, you can check that you get a Message. when there is a snippet, you can test you get a Table, this way we'll ensure there is an HTML representation and a plain-text representation

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/tests/test_magic_cmd.py [hunk 2]
```diff
@@ -416,9 +416,25 @@ def test_test_error(ip, cell, error_type, error_message):
...
+    if precmd:
+        ip_snippets.run_cell(precmd)
+    out = str(ip_snippets.run_cell("%sqlcmd snippets").result)
+    for snippet in valid_snippets:
```
[the remark this answers is not in the document]
Comment (bbeat2782) [reply src/tests/test_magic_cmd.py:435]: Makes more sense! Will make the change.

File: src/tests/test_magic_cmd.py [hunk 3]
```diff
@@ -416,9 +417,29 @@ def test_test_error(ip, cell, error_type, error_message):
...
+    [
+        (
+            "",
+            Table(
+                ["Stored snippets"],
+                [["high_price"], ["high_price_a"], ["high_price_b"]],
+            ),
+        ),
+        (
+            "%sqlcmd snippets -d high_price_a",
```
Comment (yafimvo) [inline src/tests/test_magic_cmd.py:431]: Deleting a snippet might affect some other tests. To make sure we are not, I think you can create a new snippet and see if it's in the table. then, clean up these snippets. I would test the following cases: no snippets, 1, 2, and 3 snippets.

**Review by yafimvo — CHANGES_REQUESTED** [review]
looks good. Added one comment about the test.

File: src/tests/test_magic_cmd.py (hunk 3 above)
Comment (bbeat2782) [reply src/tests/test_magic_cmd.py]: @yafimvo 
To ensure it doesn't affect other tests, I created a new `@pytest.fixture` for this test function for creating and deleting snippets.

**Review by yafimvo — APPROVED** [review]
lgtm!

**Review by edublancas — APPROVED** [review]

**Pull request merged**